cli/src/commands/operation/revert.rs

```diff
@@ -46,14 +46,23 @@ pub struct OperationRevertArgs {
     pub(crate) what: Vec<RevertWhatToRestore>, // pub for `jj undo`
 }
 
-pub(crate) fn tx_description(op: &Operation) -> String {
+fn tx_description(op: &Operation) -> String {
     format!("revert operation {}", op.id().hex())
 }
 
 pub fn cmd_op_revert(
     ui: &mut Ui,
     command: &CommandHelper,
     args: &OperationRevertArgs,
+) -> Result<(), CommandError> {
+    cmd_op_revert_with_tx_description(ui, command, args, tx_description)
+}
+
+pub(crate) fn cmd_op_revert_with_tx_description(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    args: &OperationRevertArgs,
+    tx_description: fn(op: &Operation) -> String,
```

Comment:
nit: `impl FnOnce(&Operation) -> String` as usual?

I think it's also good to duplicate the implementation.